pkg/eestream/rs_test.go

```diff
@@ -640,3 +644,37 @@ func BenchmarkReedSolomonErasureScheme(b *testing.B) {
 		}
 	}
 }
+
+func TestCalcPieceSize(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	for i, dataSize := range []int64{
+		0,
+		1,
+		1*memory.KiB.Int64() - uint32Size,
+		1 * memory.KiB.Int64(),
+		32*memory.KiB.Int64() - uint32Size,
+		32 * memory.KiB.Int64(),
+		32*memory.KiB.Int64() + 100,
+	} {
+		errTag := fmt.Sprintf("%d. %+v", i, dataSize)
+
+		fc, err := infectious.NewFEC(2, 4)
+		require.NoError(t, err, errTag)
+		es := NewRSScheme(fc, 1*memory.KiB.Int())
+		rs, err := NewRedundancyStrategy(es, 0, 0)
+		require.NoError(t, err, errTag)
+
+		calculatedSize := CalcPieceSize(int64(dataSize), es)
+
+		readers, err := EncodeReader(ctx, PadReader(ioutil.NopCloser(io.LimitReader(cryptorand.Reader, dataSize)), es.StripeSize()), rs)
```

Comment:
Do we care that we have a random data being encrypted? We could just have `InfiniteReader{}` that's used here.

Reply:
Which package is `InfiniteReader` in?

Reply:
```
type InfiniteReader struct {}
func (*InfiniteReader) Read(data []byte) (int, error) { return len(data), nil }
```

Reply:
Oh. You meant I should define one...
Nevertheless, I think it is better with the random data instead of all-zeros byte slice, or any other pattern.